spec/classes/rabbitmq_spec.rb

```diff
@@ -144,22 +141,19 @@
             it { is_expected.not_to contain_file('/etc/default/rabbitmq-server') }
           end
 
-          if has_systemd
+          if facts[:systemd]
             it do
-              is_expected.to contain_file("/etc/systemd/system/#{packagename}.service.d/limits.conf").
-                with_owner('0').
-                with_group('0').
-                with_mode('0644').
-                that_notifies('Exec[rabbitmq-systemd-reload]').
-                with_content("[Service]\nLimitNOFILE=#{value}\n")
+              is_expected.to contain_systemd__service_limits('rabbitmq.service').
+                with_limits('LimitNOFILE' => value).
+                with_restart_service(false)
             end
           else
-            it { is_expected.not_to contain_file('/etc/systemd/system/rabbitmq-server.service.d/limits.conf') }
+            it { is_expected.not_to contain_systemd__service_limits('rabbitmq.service') }
           end
         end
       end
 
-      [-42, '-42', 'foo', '42'].each do |value|
+      [-42, '-42', 'foo'].each do |value|
         context "with file_limit => '#{value}'" do
```

Comment:
Are positive numbers not supported (either as string or int) here?

Reply:
it took me a three times to read this code correctly, but this array lists unsupported values. quoted strings are now allowed.